e2e/testdata/fn-eval/include-metadata/Kptfile

```diff
@@ -0,0 +1,4 @@
+apiVersion: kpt.dev/v1alpha2
+kind: Kptfile
+metadata:
+  name: app
```

Comment:
add a test case that has a function config file

Reply:
Ignoring function config files used in pipeline haven't been implemented. Will add the tests with the implementation

Reply:
https://github.com/GoogleContainerTools/kpt/issues/1670

Reply:
https://en.wikipedia.org/wiki/Test-driven_development

You can add a test case now and mark it skipped if it faililng.

Reply:
Actually why is this PR not doing that? PR should be an atomic unit of work, having this flag without including function config is not useful on its own.

Reply:
There are 2 kinds of function config files:

1. The function config file used by the imperative function. This file will be included in function input if it's in the package and not included if it isn't as we discussed before. If we add `--include-meta-resources` flag, should we include it even it's not in the package?
2. The config files used by pipeline. Current implementation of `fn eval` include them by default. If I understand correctly this needs to be changed (#1670). They will be included if `--include-meta-resources` flag is given. I am ok to do it in this PR.

Reply:
When we say mutating a function config, it's always the function config referenced by the pipeline. The function config passed as flag to `eval` is irrelevant to this discussion.

Definition of `meta resource`: Kptfile and any referenced function configs. These resources are only for consumption of the kpt tool and will not be deployed to the cluster. There may be other kind of such resources, so there should be a single place in the codebase defining it.

Reply:
We completely missed the functionality of ignoring the function configs in `fn eval`. `fn eval` has mostly been pipeline unaware so far. I suspect this may turn out to be a bigger change because it will change the way we gather resources in `fn eval`. `pkg.Pkg` abstraction provides a method `localResources(inclueMetaResources:false)`, so that will be very useful in this case.

Reply:
Here is the problem. `fn eval` uses `kio.LocalPackageReaderWriter`. If we filter out the function config files, it will remove the files. If we want to avoid this, we need to set `NoDeleteFiles` and that will keep all files even the resources in a file are deleted by function. We need to add a fine-grained ignore matcher to the `kio.LocalPackageReaderWriter`. I suggest to add the test case but make it skipped and implement it in future PR.

Current status:
1. `Kptfile` is controlled by `--include-meta-resources`
2. Function config files are not.

Reply:
Here is what we can do:

1. Add e2e testcases for `fn eval`. Mark the ones as skipped not supported (or expect them to fail).
2. Restructure the code to support the `excluding function configs`
3. Enable tests for function config.

Good test coverage will help in refactoring the code.